src/csi/mod.rs

```diff
@@ -418,81 +397,88 @@ mod test {
         let driver_name = util::CSI_PLUGIN_NAME.to_owned();
         let data_dir = DATA_DIR.to_owned();
         let etcd_address_vec = get_etcd_address_vec();
-        let etcd_delegate = EtcdDelegate::new(etcd_address_vec)?;
 
-        let async_server = true;
         GRPC_SERVER.call_once(move || {
-            let clear_res = clear_test_data(&etcd_delegate);
-            assert!(
-                clear_res.is_ok(),
-                "failed to clear test data, the error is: {}",
-                clear_res.unwrap_err(),
-            );
-            let controller_meta_data = match build_meta_data(
-                worker_port,
-                node_id.clone(),
-                ip_address,
-                data_dir.clone(),
-                RunAsRole::Controller,
-                etcd_delegate.clone(),
-            ) {
-                Ok(md) => md,
-                Err(e) => panic!("failed to build meta data, the error is : {}", e,),
-            };
-            let controller_md = Arc::new(controller_meta_data);
-            let controller_server = match build_grpc_controller_server(
-                &controller_end_point,
-                &driver_name,
-                Arc::<MetaData>::clone(&controller_md),
-            ) {
-                Ok(server) => server,
-                Err(e) => panic!("failed to build CSI server, the error is : {}", e,),
-            };
-            let node_meta_data = match build_meta_data(
-                worker_port,
-                node_id.clone(),
-                ip_address,
-                data_dir.clone(),
-                RunAsRole::Node,
-                etcd_delegate.clone(),
-            ) {
-                Ok(md) => md,
-                Err(e) => panic!("failed to build meta data, the error is : {}", e,),
-            };
-            let node_worker_md = Arc::new(node_meta_data);
-            let node_server = match build_grpc_node_server(
-                &node_end_point,
-                &driver_name,
-                Arc::<MetaData>::clone(&node_worker_md),
-            ) {
-                Ok(server) => server,
-                Err(e) => panic!("failed to build Node server, the error is : {}", e,),
-            };
-            let worker_server = match build_grpc_worker_server(node_worker_md) {
-                Ok(server) => server,
-                Err(e) => panic!("failed to build Worker server, the error is : {}", e,),
-            };
-
-            // Keep running the task in the background
-            let _controller_thread = thread::spawn(move || {
-                run_grpc_servers(&mut [controller_server], async_server);
-            });
-
-            // Keep running the task in the background
-            let _node_thread = thread::spawn(move || {
-                run_grpc_servers(&mut [node_server, worker_server], async_server);
+            futures::executor::block_on(async {
```

Comment:
Is it possible to use tokio::spawn here?

Reply:
done